### PR TITLE
AWS::Config::ConfigurationRecorder.ResourceTypes AllowedValues

### LINKED ISSUE #1491: Missing AWS::Config::ConfigurationRecorder.ResourceTypes
*cfn-lint version: (`cfn-lint --version`)* 0.29.6

*Description of issue.*
I tried to deploy a ConfigRule, and 6 resource types that I'm able to list through the AWS UI and CLI aren't passing the linting:
```
        "AWS::ApiGateway::DomainName",
        "AWS::ApiGateway::Method",
        "AWS::ApiGatewayV2::DomainName",
        "AWS::CertificateManager::Certificate",
        "AWS::SNS::Topic",
        "AWS::SecretsManager::Secret"
```

Unfortunately, there is no exhaustive list of what's allowed via AWS::Config::ConfigRule
https://docs.aws.amazon.com/config/latest/developerguide/required-tags.html is a subset, but not exhaustive.

Looks like another case of specs lagging behind the actual AWS releases (#1219)

### PR DESCRIPTION
**Pull request by alexmlee**
fixes #1491 

*Description of changes:*
Add new resource types to US-west-2

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by alexmlee** [comment]
> Thanks for the contribution! [Left a note on the issue if you're interesting in getting these fixes into all regions](https://github.com/aws-cloudformation/cfn-python-lint/issues/1491#issuecomment-619195494)

I updated the PR to have change the file the autogeneration stems from. Do I need to issue the command to auto-generate the region files and commit those as well?

Thanks!

File: src/cfnlint/data/ExtendedSpecs/all/03_value_types/aws_config.json [hunk 1]
```diff
@@ -5,14 +5,18 @@
     "value": {
       "AllowedValues": [
         "AWS::ACM::Certificate",
+        "AWS::ApiGateway::DomainName",
+        "AWS::ApiGateway::Method",
         "AWS::ApiGateway::RestApi",
         "AWS::ApiGateway::Stage",
+        "AWS::ApiGatewayV2::DomainName",
```
Comment (PatMyron) [inline src/cfnlint/data/ExtendedSpecs/all/03_value_types/aws_config.json:12]: [not seeing these `AWS::ApiGateway` additions documented here](https://docs.aws.amazon.com/config/latest/developerguide/resource-config-reference.html#amazonapigateway) or in the [Config console](https://console.aws.amazon.com/config/home#/first-run/configuration)

Do you know if they are recent additions or if there is somewhere else to find them?

Screen Shot 2020-04-24 at 3 57 11 PM

File: src/cfnlint/data/ExtendedSpecs/all/03_value_types/aws_config.json [hunk 2]
```diff
@@ -5,14 +5,18 @@
...
         "AWS::ApiGatewayV2::Api",
         "AWS::ApiGatewayV2::Stage",
         "AWS::AutoScaling::AutoScalingGroup",
         "AWS::AutoScaling::LaunchConfiguration",
         "AWS::AutoScaling::ScalingPolicy",
         "AWS::AutoScaling::ScheduledAction",
+        "AWS::CertificateManager::Certificate",
```
Comment (PatMyron) [inline src/cfnlint/data/ExtendedSpecs/all/03_value_types/aws_config.json:19]: Do you know if this one should be [`AWS::ACM::Certificate`](https://docs.aws.amazon.com/config/latest/developerguide/resource-config-reference.html#awscertificatemanager) or [`AWS::CertificateManager::Certificate`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-certificatemanager-certificate.html)?

**Comment by PatMyron** [comment]
> I updated the PR to have change the file the autogeneration stems from. Do I need to issue the command to auto-generate the region files and commit those as well?

Source file should be enough, we can handle the autogeneration!

File: src/cfnlint/data/ExtendedSpecs/all/03_value_types/aws_config.json (hunk 1 above)
Comment (alexmlee) [reply src/cfnlint/data/ExtendedSpecs/all/03_value_types/aws_config.json:12]: 🤔 Huh! I swear they were in here last week.

File: src/cfnlint/data/ExtendedSpecs/all/03_value_types/aws_config.json (hunk 2 above)
Comment (alexmlee) [reply src/cfnlint/data/ExtendedSpecs/all/03_value_types/aws_config.json:19]: This one has also disappeared in the last week as an option. I think it was replaced by AWS::ACM::Certificate.

**Comment by PatMyron** [comment]
@alexmlee should we merge these in? Or do you want to keep looking into the other resource types?

**Comment by alexmlee** [comment]
> @alexmlee should we merge these in? Or do you want to keep looking into the other resource types?

I think this is ready to go!

**Comment by PatMyron** [comment]
@alexmlee [looking into starting to replace these manually maintaining AllowedValues with `botocore` `enums`](https://github.com/aws-cloudformation/cfn-python-lint/issues/50#issuecomment-668265454) and the [`botocore` `enum` for this](https://github.com/boto/botocore/blob/ef0aa3ea8e010e064fdd0a6c6634eb5ad7b27429/botocore/data/config/2014-11-12/service-2.json#L5972-L6066) actually does not contain these two resource types but [they do seem to be documented](https://docs.aws.amazon.com/config/latest/developerguide/resource-config-reference.html) 🤔

**Pull request merged**